libfwupdplugin/fu-kernel.c

```diff
@@ -305,3 +305,108 @@ fu_kernel_get_cmdline(GError **error)
 	return NULL;
 #endif
 }
+
+gboolean
+fu_kernel_is_cmdline_mutable(void)
+{
+	const gchar *config_files[] = {"/boot/grub2/grub.cfg", "/etc/grub2.cfg", NULL};
+	g_autofree gchar *grubby_path = fu_path_find_program("grubby", NULL);
+
+	/* not found */
+	if (grubby_path == NULL)
+		return FALSE;
+
+	/* check all the config files are writable */
+	for (guint i = 0; config_files[i] != NULL; i++) {
+		g_autoptr(GFile) file = g_file_new_for_path(config_files[i]);
+		g_autoptr(GFileInfo) info = NULL;
+		g_autoptr(GError) error_local = NULL;
```

Comment:
you don't think a new helper for this that both could call could work?  It seems like the same to me.  Am I missing some nuance?

Reply:
I had something like `gboolean fu_common_file_is_writable(GFile *file, gboolean *is_writable, GError **error)` and it felt much clumsier than just copy and pasting the ~8 lines of code.

Reply:
I've got something else I'm working on that may use this same function, so it will be 3 places that use the same 8 lines.  Still copy/paste better?

Reply:
What I was struggling with was "should a file that doesn't exist be a GError" and "should a file that's not writeable be an error" -- the caller sometimes wants the former and sometimes wants the latter...